### ISSUE TITLE
SWITCH_DEVICES flag not working as expected

### ISSUE THREAD

**Issue by anjackson** [ukwa/webrender-puppeteer#7]
Deployed `1.0.4` on live and got a lot of

```
WARNING:root:FAILED:\logs=b"Browser arguments:  { ignoreHTTPSErrors: true,\n  args: \n   [ '--disk-cache-size=0',\n     '--no-sandbox',\n     '--igno
re-certificate-errors',\n     '--proxy-server=warcprox:8000' ] }\nNavigating to https://twitter.com/WYSdaily/\nScrolling down...\nLooking for any mod
al buttons...\nWaiting for any activity to die down...\nRendering...\nunhandledRejection:  switchDevices is not defined Promise {\n  <rejected> Refer
enceError: switchDevices is not defined\n    at /renderer.js:168:3\n    at <anonymous>\n    at process._tickCallback (internal/process/next_tick.js:1
89:7) }\n"
```

So `switchDevices` was not getting defined, not clear why.

Should also check the output override feature is working as expected.

**Comment by anjackson** [comment]
Additional issue:

```
fc_crawl_webrender.1.9gckrmv69lwd@crawler05    | INFO:flask.app:Got URL: https://twitter.com/hashtag/Corònabhìoras/
fc_crawl_webrender.1.9gckrmv69lwd@crawler05    | [2020-06-01 11:19:55,245] ERROR in app: Exception on /render [GET]
fc_crawl_webrender.1.9gckrmv69lwd@crawler05    | Traceback (most recent call last):
fc_crawl_webrender.1.9gckrmv69lwd@crawler05    |   File "/usr/local/lib/python3.7/site-packages/flask/app.py", line 2292, in wsgi_app
fc_crawl_webrender.1.9gckrmv69lwd@crawler05    |     response = self.full_dispatch_request()
fc_crawl_webrender.1.9gckrmv69lwd@crawler05    |   File "/usr/local/lib/python3.7/site-packages/flask/app.py", line 1815, in full_dispatch_request
fc_crawl_webrender.1.9gckrmv69lwd@crawler05    |     rv = self.handle_user_exception(e)
fc_crawl_webrender.1.9gckrmv69lwd@crawler05    |   File "/usr/local/lib/python3.7/site-packages/flask/app.py", line 1718, in handle_user_exception
fc_crawl_webrender.1.9gckrmv69lwd@crawler05    |     reraise(exc_type, exc_value, tb)
fc_crawl_webrender.1.9gckrmv69lwd@crawler05    |   File "/usr/local/lib/python3.7/site-packages/flask/_compat.py", line 35, in reraise
fc_crawl_webrender.1.9gckrmv69lwd@crawler05    |     raise value
fc_crawl_webrender.1.9gckrmv69lwd@crawler05    |   File "/usr/local/lib/python3.7/site-packages/flask/app.py", line 1813, in full_dispatch_request
fc_crawl_webrender.1.9gckrmv69lwd@crawler05    |     rv = self.dispatch_request()
fc_crawl_webrender.1.9gckrmv69lwd@crawler05    |   File "/usr/local/lib/python3.7/site-packages/flask/app.py", line 1799, in dispatch_request
fc_crawl_webrender.1.9gckrmv69lwd@crawler05    |     return self.view_functions[rule.endpoint](**req.view_args)
fc_crawl_webrender.1.9gckrmv69lwd@crawler05    |   File "/webrender/webrender/wrengine.py", line 59, in render
fc_crawl_webrender.1.9gckrmv69lwd@crawler05    |     include_rendered=include_rendered,  target_date=target_date))
fc_crawl_webrender.1.9gckrmv69lwd@crawler05    |   File "/webrender/webrender/puppeteer/docker.py", line 83, in get_har_with_image
fc_crawl_webrender.1.9gckrmv69lwd@crawler05    |     include_rendered_in_har=include_rendered, return_screenshot=return_screenshot)
fc_crawl_webrender.1.9gckrmv69lwd@crawler05    |   File "/webrender/webrender/puppeteer/docker.py", line 181, in _warcprox_write_har_content
fc_crawl_webrender.1.9gckrmv69lwd@crawler05    |     extra_headers= warcprox_headers )
fc_crawl_webrender.1.9gckrmv69lwd@crawler05    |   File "/webrender/webrender/puppeteer/docker.py", line 268, in _warcprox_write_record
fc_crawl_webrender.1.9gckrmv69lwd@crawler05    |     with urllib.request.urlopen(request) as response:
fc_crawl_webrender.1.9gckrmv69lwd@crawler05    |   File "/usr/local/lib/python3.7/urllib/request.py", line 222, in urlopen
fc_crawl_webrender.1.9gckrmv69lwd@crawler05    |     return opener.open(url, data, timeout)
fc_crawl_webrender.1.9gckrmv69lwd@crawler05    |   File "/usr/local/lib/python3.7/urllib/request.py", line 525, in open
fc_crawl_webrender.1.9gckrmv69lwd@crawler05    |     response = self._open(req, data)
fc_crawl_webrender.1.9gckrmv69lwd@crawler05    |   File "/usr/local/lib/python3.7/urllib/request.py", line 543, in _open
fc_crawl_webrender.1.9gckrmv69lwd@crawler05    |     '_open', req)
fc_crawl_webrender.1.9gckrmv69lwd@crawler05    |   File "/usr/local/lib/python3.7/urllib/request.py", line 503, in _call_chain
fc_crawl_webrender.1.9gckrmv69lwd@crawler05    |     result = func(*args)
fc_crawl_webrender.1.9gckrmv69lwd@crawler05    |   File "/usr/local/lib/python3.7/urllib/request.py", line 1347, in http_open
fc_crawl_webrender.1.9gckrmv69lwd@crawler05    |     return self.do_open(http.client.HTTPConnection, req)
fc_crawl_webrender.1.9gckrmv69lwd@crawler05    |   File "/usr/local/lib/python3.7/urllib/request.py", line 1319, in do_open
fc_crawl_webrender.1.9gckrmv69lwd@crawler05    |     encode_chunked=req.has_header('Transfer-encoding'))
fc_crawl_webrender.1.9gckrmv69lwd@crawler05    |   File "/usr/local/lib/python3.7/http/client.py", line 1252, in request
fc_crawl_webrender.1.9gckrmv69lwd@crawler05    |     self._send_request(method, url, body, headers, encode_chunked)
fc_crawl_webrender.1.9gckrmv69lwd@crawler05    |   File "/usr/local/lib/python3.7/http/client.py", line 1263, in _send_request
fc_crawl_webrender.1.9gckrmv69lwd@crawler05    |     self.putrequest(method, url, **skips)
fc_crawl_webrender.1.9gckrmv69lwd@crawler05    |   File "/usr/local/lib/python3.7/http/client.py", line 1118, in putrequest
fc_crawl_webrender.1.9gckrmv69lwd@crawler05    |     self._output(self._encode_request(request))
fc_crawl_webrender.1.9gckrmv69lwd@crawler05    |   File "/usr/local/lib/python3.7/http/client.py", line 1198, in _encode_request
fc_crawl_webrender.1.9gckrmv69lwd@crawler05    |     return request.encode('ascii')
fc_crawl_webrender.1.9gckrmv69lwd@crawler05    | UnicodeEncodeError: 'ascii' codec can't encode character '\xf2' in position 64: ordinal not in range(128)
fc_crawl_webrender.1.9gckrmv69lwd@crawler05    | ERROR:flask.app:Exception on /render [GET]
fc_crawl_webrender.1.9gckrmv69lwd@crawler05    | Traceback (most recent call last):
fc_crawl_webrender.1.9gckrmv69lwd@crawler05    |   File "/usr/local/lib/python3.7/site-packages/flask/app.py", line 2292, in wsgi_app
fc_crawl_webrender.1.9gckrmv69lwd@crawler05    |     response = self.full_dispatch_request()
fc_crawl_webrender.1.9gckrmv69lwd@crawler05    |   File "/usr/local/lib/python3.7/site-packages/flask/app.py", line 1815, in full_dispatch_request
fc_crawl_webrender.1.9gckrmv69lwd@crawler05    |     rv = self.handle_user_exception(e)
fc_crawl_webrender.1.9gckrmv69lwd@crawler05    |   File "/usr/local/lib/python3.7/site-packages/flask/app.py", line 1718, in handle_user_exception
fc_crawl_webrender.1.9gckrmv69lwd@crawler05    |     reraise(exc_type, exc_value, tb)
fc_crawl_webrender.1.9gckrmv69lwd@crawler05    |   File "/usr/local/lib/python3.7/site-packages/flask/_compat.py", line 35, in reraise
fc_crawl_webrender.1.9gckrmv69lwd@crawler05    |     raise value
fc_crawl_webrender.1.9gckrmv69lwd@crawler05    |   File "/usr/local/lib/python3.7/site-packages/flask/app.py", line 1813, in full_dispatch_request
fc_crawl_webrender.1.9gckrmv69lwd@crawler05    |     rv = self.dispatch_request()
fc_crawl_webrender.1.9gckrmv69lwd@crawler05    |   File "/usr/local/lib/python3.7/site-packages/flask/app.py", line 1799, in dispatch_request
fc_crawl_webrender.1.9gckrmv69lwd@crawler05    |     return self.view_functions[rule.endpoint](**req.view_args)
fc_crawl_webrender.1.9gckrmv69lwd@crawler05    |   File "/webrender/webrender/wrengine.py", line 59, in render
fc_crawl_webrender.1.9gckrmv69lwd@crawler05    |     include_rendered=include_rendered,  target_date=target_date))
fc_crawl_webrender.1.9gckrmv69lwd@crawler05    |   File "/webrender/webrender/puppeteer/docker.py", line 83, in get_har_with_image
fc_crawl_webrender.1.9gckrmv69lwd@crawler05    |     include_rendered_in_har=include_rendered, return_screenshot=return_screenshot)
fc_crawl_webrender.1.9gckrmv69lwd@crawler05    |   File "/webrender/webrender/puppeteer/docker.py", line 181, in _warcprox_write_har_content
fc_crawl_webrender.1.9gckrmv69lwd@crawler05    |     extra_headers= warcprox_headers )
fc_crawl_webrender.1.9gckrmv69lwd@crawler05    |   File "/webrender/webrender/puppeteer/docker.py", line 268, in _warcprox_write_record
fc_crawl_webrender.1.9gckrmv69lwd@crawler05    |     with urllib.request.urlopen(request) as response:
fc_crawl_webrender.1.9gckrmv69lwd@crawler05    |   File "/usr/local/lib/python3.7/urllib/request.py", line 222, in urlopen
fc_crawl_webrender.1.9gckrmv69lwd@crawler05    |     return opener.open(url, data, timeout)
fc_crawl_webrender.1.9gckrmv69lwd@crawler05    |   File "/usr/local/lib/python3.7/urllib/request.py", line 525, in open
fc_crawl_webrender.1.9gckrmv69lwd@crawler05    |     response = self._open(req, data)
fc_crawl_webrender.1.9gckrmv69lwd@crawler05    |   File "/usr/local/lib/python3.7/urllib/request.py", line 543, in _open
fc_crawl_webrender.1.9gckrmv69lwd@crawler05    |     '_open', req)
fc_crawl_webrender.1.9gckrmv69lwd@crawler05    |   File "/usr/local/lib/python3.7/urllib/request.py", line 503, in _call_chain
fc_crawl_webrender.1.9gckrmv69lwd@crawler05    |     result = func(*args)
fc_crawl_webrender.1.9gckrmv69lwd@crawler05    |   File "/usr/local/lib/python3.7/urllib/request.py", line 1347, in http_open
fc_crawl_webrender.1.9gckrmv69lwd@crawler05    |     return self.do_open(http.client.HTTPConnection, req)
fc_crawl_webrender.1.9gckrmv69lwd@crawler05    |   File "/usr/local/lib/python3.7/urllib/request.py", line 1319, in do_open
fc_crawl_webrender.1.9gckrmv69lwd@crawler05    |     encode_chunked=req.has_header('Transfer-encoding'))
fc_crawl_webrender.1.9gckrmv69lwd@crawler05    |   File "/usr/local/lib/python3.7/http/client.py", line 1252, in request
fc_crawl_webrender.1.9gckrmv69lwd@crawler05    |     self._send_request(method, url, body, headers, encode_chunked)
fc_crawl_webrender.1.9gckrmv69lwd@crawler05    |   File "/usr/local/lib/python3.7/http/client.py", line 1263, in _send_request
fc_crawl_webrender.1.9gckrmv69lwd@crawler05    |     self.putrequest(method, url, **skips)
fc_crawl_webrender.1.9gckrmv69lwd@crawler05    |   File "/usr/local/lib/python3.7/http/client.py", line 1118, in putrequest
fc_crawl_webrender.1.9gckrmv69lwd@crawler05    |     self._output(self._encode_request(request))
fc_crawl_webrender.1.9gckrmv69lwd@crawler05    |   File "/usr/local/lib/python3.7/http/client.py", line 1198, in _encode_request
fc_crawl_webrender.1.9gckrmv69lwd@crawler05    |     return request.encode('ascii')
fc_crawl_webrender.1.9gckrmv69lwd@crawler05    | UnicodeEncodeError: 'ascii' codec can't encode character '\xf2' in position 64: ordinal not in range(128)
fc_crawl_webrender.1.9gckrmv69lwd@crawler05    | 10.0.6.7 - - [01/Jun/2020:11:19:55 +0000] "GET /render?url=https://twitter.com/hashtag/Cor%C3%B2nabh%C3%ACora
s/&warc_prefix=BL-NPLD-WEBRENDER-frequent-npld-20200529084630 HTTP/1.1" 500 290 "-" "Java/1.8.0_232"
```

**Comment by anjackson** [comment]
Feature was not creating variables with the right scope. Fixed in 5e8c1305864f0b0516826cffd15adc6d3f725de6
